BookController and dto implemented

jszymanski97 opened the pull request:
BookController.java - created with supporting BookDto.java and CreateBookRequestDto.java classes
MapperConfig.java - implemented mapstruct to ease the class conversion
EntityNotFoundException.java - exception created for error message clarity

jszymanski97 committed:
BookController.java - created with supporting BookDto.java and CreateBookRequestDto.java classes
MapperConfig.java - implemented mapstruct to ease the class conversion
EntityNotFoundException.java - exception created for error message clarity

jszymanski97 committed:
Revert "BookController.java - created with supporting BookDto.java and CreateBookRequestDto.java classes"

This reverts commit 8fea99e15152627343f2f06117c7e2a9b2fa9fcc.

jszymanski97 committed:
Revert "Revert "BookController.java - created with supporting BookDto.java and CreateBookRequestDto.java classes""

This reverts commit 5885b87b52ef49256948b0c1a062bd7e6efc07ed.

MateuszRostek commented on src/main/java/mate/project/dto/BookDto.java:
```diff
@@ -0,0 +1,16 @@
+
+@Data
+public class BookDto {
+
```
Do not add redundant empty spaces.

MateuszRostek commented on src/main/java/mate/project/dto/CreateBookRequestDto.java:
```diff
@@ -0,0 +1,15 @@
+
+@Data
+public class CreateBookRequestDto {
+
```
Do not add redundant empty spaces.

MateuszRostek commented on src/main/java/mate/project/dto/CreateBookRequestDto.java:
```diff
@@ -0,0 +1,15 @@
+    private BigDecimal price;
+    private String description;
+    private String coverImage;
+}
```
Make it a `record`.

MateuszRostek commented on src/main/java/mate/project/mapper/BookMapper.java:
```diff
@@ -0,0 +1,15 @@
+
+@Mapper(config = MapperConfig.class)
+public interface BookMapper {
+
```
Do not add redundant empty spaces.

MateuszRostek commented on src/main/java/mate/project/repository/BookRepository.java:
```diff
@@ -7,4 +7,6 @@ public interface BookRepository {
 
     List<Book> findAll();
+
+    Book getBookById(Long id);
```
You should use Optional here:
```suggestion
    Optional<Book> getBookById(Long id);
```

jszymanski97 replied:
Okay, I went by the video where the coach explained to use optional for methods like findBookById where we not sure if the book exists, and for methods with get we dont need to. But I see and understand why it is better to use optional for both cases

MateuszRostek commented on src/main/java/mate/project/service/BookServiceImpl.java:
```diff
@@ -1,28 +1,38 @@
+@RequiredArgsConstructor
 @Service
 public class BookServiceImpl implements BookService {
+
```
Do not add redundant empty spaces.

MateuszRostek commented on src/main/java/mate/project/service/BookServiceImpl.java:
```diff
@@ -1,28 +1,38 @@
+    @Override
+    public BookDto save(CreateBookRequestDto createBookRequestDto) {
+        Book model = bookMapper.toModel(createBookRequestDto);
+        Book savedModel = bookRepository.save(model);
+        return bookMapper.toBookDto(savedModel);
```
```suggestion
        return bookMapper.toBookDto(bookRepository.save(model));
```

MateuszRostek commented on src/main/resources/application.properties:
Add this one also:

`spring.jpa.open-in-view=false`

MateuszRostek reviewed (changes requested):
Nice try! See some advice and implement required changes.

jszymanski97 committed:
redundant empty lines removed,
CreateBookRequestDto.java changed in to a record
getBookById method now returns Optional<Book>

MateuszRostek commented:
In general I do not recommend to use and create methods of type `get`, it is safer to create methods of type `find` and return Optional from them:)

MateuszRostek reviewed (approved):
Good job!

MateuszRostek commented:
Remember to merge this branch and create next solution on the new one with new pull request:)
